### PR TITLE
Add a basic script for Pandoc

### PR DESCRIPTION
**Pull request by amlmn**

### CODE REVIEW

File: .style.yapf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[style]
+based_on_style: google
+column_limit: 80
```
Comment (wbthomason) [inline .style.yapf:3]: Ew. 100 is better.

(Mostly joking; I prefer 100 for my stuff but it's super trivial and I don't actually care here.)

File: regenerate.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--posts',
+        type=str,
+        default='algos',
+        help='directory relative to the repo root')
+    parser.add_argument(
+        '--pandoc_flags', type=str, help='passed verbatim to pandoc')
```
Comment (wbthomason) [inline regenerate.py]: There's presumably a set of default flags we'll want to apply?

File: regenerate.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+
+def run(*args) -> str:
+    return subprocess.run(
+        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='utf-8')
+
+
+def GatherPosts(posts_dir: str) -> List[str]:
+    post_paths = run('find', posts_dir, '-name',
+                     '*.md').stdout.strip().split('\n')
```
Comment (wbthomason) [inline regenerate.py]: Why not just use glob?

File: regenerate.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    return post_paths
+
+
+def GenerateHtml(pandoc_flags: str, post_paths: List[str]):
+    for post_path in post_paths:
+        output_path = post_path[:-2] + 'html'
```
Comment (wbthomason) [inline regenerate.py]: I think we should place output in a single root-level directory to make it easier to build the whole site.

This also doesn't handle copying figures or JS, or generating pages like the index or category lists. Maybe that's out of scope for this PR?

File: regenerate.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        run('pandoc', '-s', pandoc_flags
```
Comment (wbthomason) [inline regenerate.py]: Definitely premature optimization, but we should eventually await these from a separate greenlet/thread each to make it faster to build a bunch of paths.

**Review by wbthomason — CHANGES_REQUESTED** [review]
This is ok, but I think path finding could be cleaner (and more platform independent) using the glob module, and we should define a default value for the flags to pass to pandoc. Also, all output should be placed in a single directory so we don't have to collect everything.

File: regenerate.py (hunk 2 above)
Comment (amlmn) [reply regenerate.py]: The plan is to add that later. Right now, I'm not confident enough about what they'll be.

File: regenerate.py (hunk 3 above)
Comment (amlmn) [reply regenerate.py]: Done. I forgot about that module.

File: regenerate.py (hunk 4 above)
Comment (amlmn) [reply regenerate.py]: > I think we should place output in a single root-level directory to make it easier to build the whole site.

Done.
> 
> This also doesn't handle copying figures or JS, or generating pages like the index or category lists. Maybe that's out of scope for this PR?

I'd prefer to do that in a separate PR.

File: regenerate.py (hunk 5 above)
Comment (amlmn) [reply regenerate.py]: Added a TODO.

File: html/example.html [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml" lang="" xml:lang="">
+<head>
+  <meta charset="utf-8" />
+  <meta name="generator" content="pandoc" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
+  <title>example</title>
+  <style type="text/css">
```
Comment (wbthomason) [inline html/example.html]: What is this file for? Can it not use in-file CSS?

**Review by wbthomason — CHANGES_REQUESTED** [review]
Looks good to me except that I'm not sure what the HTML file is and would like it to not use in-file CSS. Why wouldn't it use `css/styles.css` or similar.

File: html/example.html (hunk 6 above)
Comment (amlmn) [reply html/example.html]: It's just pandoc's default. We'll fix it later.

The commit is really just supposed to be for a basic python script.

**Review by wbthomason — APPROVED** [review]
Looks great, thank you!

**Pull request merged**